### PR TITLE
[BugFix] Fix disable base compaction with minute granularity & fe/be recover

### PR DESCRIPTION
**Pull request by meegoo**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

File: test/sql/test_compaction_cron/R/test_compaction_cron [hunk 1]
```diff
@@ -1,4 +1,72 @@
...
+create database test_disable_compaction;
+-- result:
+-- !result
+use test_disable_compaction;
+-- result:
+-- !result
+create table t(k int, v int) properties('base_compaction_forbidden_time_ranges'='* * * * *');
+-- result:
+-- !result
+update information_schema.be_configs set value='-1' where name='disable_compaction_tablet_size_threshold';
```
Comment (kevincai) [inline test/sql/test_compaction_cron/R/test_compaction_cron:11]: need to reset it to default value, avoid affecting subsequent cases even the sql tester is running in sequential.

File: be/src/common/config.h [hunk 2]
```diff
@@ -379,6 +379,10 @@ CONF_mInt64(size_tiered_level_multiple, "5");
 CONF_mInt64(size_tiered_level_multiple_dupkey, "10");
 CONF_mInt64(size_tiered_level_num, "7");
 
+// disable compaction will take affect when tablet size is larger than this value
```
Comment (kevincai) [inline be/src/common/config.h:382]: skip the compaction against the tablets whose size is larger than this value.

**Review by gengjun-git — DISMISSED** [review]

File: be/src/storage/compaction_manager.cpp [hunk 3]
```diff
@@ -200,6 +203,27 @@ void CompactionManager::remove_candidate(int64_t tablet_id) {
     }
 }
 
+bool CompactionManager::_check_disable_compaction(const CompactionCandidate& candidate) {
```
Comment (kevincai) [inline be/src/storage/compaction_manager.cpp:206]: _check_compaction_disabled

File: be/src/storage/tablet_meta.h [hunk 4]
```diff
@@ -135,6 +135,7 @@ class TabletMeta {
     TabletTypePB tablet_type() const { return _tablet_type; }
     TabletUid tablet_uid() const;
     int64_t table_id() const;
+    void set_table_id(int64_t table_id);
```
Comment (kevincai) [inline be/src/storage/tablet_meta.h:138]: should mark as TEST only, regular code should never use this interface to change tablet id directly.

```suggestion
    void TEST_set_table_id(int64_t table_id);
```

File: test/sql/test_compaction_cron/R/test_compaction_cron (hunk 1 above)
Comment (meegoo) [reply test/sql/test_compaction_cron/R/test_compaction_cron:11]: Restoring it to the original value does not guarantee that the parallel execution case will be executed correctly, so the only way is set this value before the execution and execute it serially.

File: be/src/common/config.h (hunk 2 above)
Comment (meegoo) [reply be/src/common/config.h:382]: It is not that all tablets larger than this value are not compacted, but when disable compaction is turned on.

Comment (kevincai) [reply be/src/common/config.h:382]: that is exactly the problem. the explanation of this parameter is not very straightforward. I misunderstood it even by looking the code. How to expect end user to understand this parameter correctly. orz

File: test/sql/test_compaction_cron/R/test_compaction_cron (hunk 1 above)
Comment (kevincai) [reply test/sql/test_compaction_cron/R/test_compaction_cron:11]: yes. it has to be set to sequential and also restore the parameter after the case is done. Be respect to the test env, leave a clean env after the case is done.

**Review by kevincai — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**